feat(BarnBridge): plugin initial support

DCarranza opened the pull request:
Part of the hackathon is to add a BarnBridge (Smart Yield) plugin to the mask network!

<!-- Please refer to the related issue number -->
https://gitcoin.co/issue/BarnBridge/Bounty-Program/14/100026336
closes #

DCarranza committed:
feat(BarnBridge): plugin initial support

Part of the hackathon is to add a BarnBridge (Smart Yield) plugin to the mask network!

DCarranza committed:
feat(BarnBridge): plugin initial support

Part of the hackathon is to add a BarnBridge (Smart Yield) plugin to the mask network!

DCarranza commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/PortfolioBalance.tsx:
```diff
@@ -0,0 +1,100 @@
+import { makeStyles } from '@masknet/theme'
+
+// import { withStyles } from "@material-ui/core/styles";
```
I ran into this being blocked by lint rules -> is there any alternative masknet prefers. makeStyles isn't really what I want here....

DCarranza committed:
fix(BarnBridge): remove useStyles

Refactor progress bars to not use useStyles as it is forbidden => use other ways to change color

septs commented on packages/maskbook/src/plugins/BarnBridge/BarnBridgeIcon.tsx:
```diff
@@ -0,0 +1,194 @@
+
+export const BarnBridgeYieldFarmingIcon: FC<SvgIconProps> = (props) => (
+    <SvgIcon {...props}>
+        <svg width="50" height="50" viewBox="0 0 50 50" fill="none" xmlns="http://www.w3.org/2000/svg">
```
plz use https://aka.pw/svgomg reduce svg image

DCarranza replied:
Will fix!

DCarranza replied:
ran them all through svgomg -> pretty awesome tool

septs commented on packages/maskbook/src/plugins/BarnBridge/Components.tsx:
```diff
@@ -0,0 +1,15 @@
+import { Box } from '@material-ui/core'
+export function HelloWorld(props: any) {
```
plz do not use `any`!

DCarranza replied:
Will remove :)

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/PortfolioBalance.tsx:
```diff
@@ -0,0 +1,96 @@
+
+type Props = {
+    total?: number
+    data: [[string, number | undefined], [string, number | undefined]]
```
plz use labeled tuple elements feature, update the type definition

see https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-0.html#labeled-tuple-elements

septs replied:
```ts
type SubItem = [label: string, value: number | undefined]
```

```suggestion
    data: [SubItem, SubItem]
```

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,143 @@
+import BigNumber from 'bignumber.js'
+
+export function getExponentValue(decimals = 0): BigNumber {
```
plz merge the file non-adhoc function to `packages/web3-shared/src/utils/number.ts`

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,143 @@
+
+    const { decimals } = options ?? {}
+
+    const val = BigNumber.isBigNumber(value) ? value.toNumber() : value
```
if this a big number over 53 bits?

see https://tc39.es/ecma262/#sec-ecmascript-language-types-number-type

DCarranza replied:
I don't believe it is so we should be OK

septs commented on packages/maskbook/src/plugins/BarnBridge/Model/SYPoolModel.tsx:
```diff
@@ -0,0 +1,64 @@
+}
+
+function PrettifyLiquidity(liqudity: string) {
+    const intLiquidity = parseInt(liqudity, 10)
```
```suggestion
    const intLiquidity = Number.parseInt(liqudity, 10)
```

septs commented on packages/maskbook/src/plugins/BarnBridge/Model/SYPoolModel.tsx:
```diff
@@ -0,0 +1,64 @@
+
+function ConstructRedirectUrl(protocolName: string, coin: string) {
+    console.log('generating url')
+    return APP_URL + SY_URL + `stats?m=${protocolName}&t=${coin}`
```
plz use https://npm.im/urlcat build the url.

septs commented on packages/maskbook/src/plugins/BarnBridge/constants.ts:
```diff
@@ -0,0 +1,28 @@
+export const API_URL = 'https://api.barnbridge.com/'
+export const SY_URL = 'smart-yield/'
+
+export const URL_PATTERN = new RegExp(`(http|https)\:\/\/.*\.?${escapeRegExp(BASE_URL)}`)
```
```suggestion
export const URL_PATTERN = new RegExp(`https?:\/\/.*\.?${escapeRegExp(BASE_URL)}`)
```

septs commented on packages/storybook-shared/index.d.ts.map:
```diff
@@ -0,0 +1 @@
+{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["index.ts"],"names":[],"mappings":"AAAA,cAAc,YAAY,CAAA"}
```
@Jack-Works need update gitignore?

Jack-Works replied:
No. This should be git tracked because sync it manually from Crowdin is much more complex

Jack-Works committed:
chore: remove extra files

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,143 @@
+        val = Number(val)
+    }
+
+    if (BigNumber.isBigNumber(val)) {
+        if (val.isNaN()) {
+            return undefined
+        }
+    } else if (typeof val === 'number') {
+        if (!Number.isFinite(val)) {
+            return undefined
+        }
+    }
```
```suggestion
    if (BigNumber.isBigNumber(val) && val.isNaN()) {
        return undefined
    } else if (typeof val === 'number' && !Number.isFinite(val)) {
        return undefined
    }
```

DCarranza committed:
feat(BarnBridge): address septs reviews

Septs reviews until Sep 14th

DCarranza commented:
Alrighty sorry for the delay! The next iteration will be faster, i promise :P

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,136 @@
+import { getExponentValue } from '@masknet/web3-shared'
+
+export function getHumanValue(value?: BigNumber, decimals = 0): BigNumber | undefined {
+    return value?.div(getExponentValue(decimals))
```
why not use `pow10`

DCarranza replied:
I grabbed this from the BarnBridge main app => most of this isn't really needed. Let me clean this up/delete

septs commented on packages/maskbook/src/plugins/BarnBridge/Model/SYPoolModel.tsx:
```diff
@@ -0,0 +1,65 @@
+
+function ConstructRedirectUrl(protocolName: string, coin: string) {
+    console.log('generating url')
+    return urlcat(APP_URL + SY_URL, `stats?m=:protocolName&t=:coin`, { protocolName: protocolName, coin: coin })
```
```suggestion
    return urlcat(APP_URL, `${SY_URL}/stats`, { m: protocolName, t: coin })
```

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,136 @@
+        return undefined
+    }
+
+    let str = ''
```
```suggestion
    let formatted = ''
```

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,136 @@
+    let str = ''
+
+    if (compact) {
+        str = Intl.NumberFormat('en', {
+            notation: 'compact',
+            maximumFractionDigits: decimals !== 0 ? decimals : undefined,
+        }).format(BigNumber.isBigNumber(val) ? val.toNumber() : val)
```
```suggestion
        const formatter = Intl.NumberFormat('en', {
            notation: 'compact',
            maximumFractionDigits: decimals !== 0 ? decimals : undefined,
        })
        formatted = formatter.format(BigNumber.isBigNumber(input) ? input.toNumber() : input)
```

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,136 @@
+    value: number | BigNumber | string | undefined,
+    options?: FormatUSDOptions,
+): string | undefined {
+    let val = value
```
```suggestion
    let input = value
```

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,136 @@
+    }
+
+    if (typeof val === 'string') {
+        val = Number(val)
```
```suggestion
        input = Number.parseInt(input, 10)
```

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,136 @@
+export function formatUSD(
+    value: number | BigNumber | string | undefined,
+    options?: FormatUSDOptions,
+): string | undefined {
+    let val = value
+
+    if (val === undefined || val === null) {
+        return undefined
+    }
+
+    if (typeof val === 'string') {
+        val = Number(val)
+    }
+
+    if (BigNumber.isBigNumber(val) && val.isNaN()) {
+        return undefined
+    } else if (typeof val === 'number' && !Number.isFinite(val)) {
+        return undefined
+    }
+
+    const { decimals = 2, compact = false } = options ?? {}
+
+    if (decimals < 0 || decimals > 20) {
+        console.trace(`Decimals value is out of range 0..20 (value: ${decimals})`)
+        return undefined
+    }
+
+    let str = ''
+
+    if (compact) {
+        str = Intl.NumberFormat('en', {
+            notation: 'compact',
+            maximumFractionDigits: decimals !== 0 ? decimals : undefined,
+        }).format(BigNumber.isBigNumber(val) ? val.toNumber() : val)
+    } else {
+        str = new BigNumber(val.toFixed(decimals)).toFormat(decimals)
+    }
+
+    return `$${str}`
+}
```
```suggestion
export function formatUSD(
    value: BigNumber.Value | undefined,
    options?: FormatUSDOptions,
): string | undefined {
    value = new BigNumber(value)
    if (!value.isFinite()) {
        return undefined
    }
    const decimals = options?.decimals ?? 2
    const compact = options?.compact ?? false
    if (decimals < 0 || decimals > 20) {
        throw new Error(`Decimals value is out of range 0..20 (value: ${decimals})`)
    }
    let formatted: string
    if (compact) {
        const formatter = Intl.NumberFormat('en', {
            notation: 'compact',
            maximumFractionDigits: decimals !== 0 ? decimals : undefined,
        })
        formatted = formatter.format(value.toNumber())
    } else {
        formatted = new BigNumber(value.toFixed(decimals)).toFormat(decimals)
    }
    return `$${formatted}`
}
```

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,136 @@
+    return `$${str}`
+}
+
+export function formatUSDValue(value?: BigNumber | number, decimals = 2, minDecimals: number = decimals): string {
```
```suggestion
export function formatUSDValue(value: BigNumber.Value, decimals = 2, minDecimals = decimals): string {
    value = new BigNumber(value)
```

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,136 @@
+    }).format(val)
+}
+
+export function formatPercent(value: number | BigNumber | undefined, decimals: number = 2): string | undefined {
```
```suggestion
export function formatPercent(value: BigNumber.Value, decimals = 2): string | undefined {
    value = new BigNumber(value)
```

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,136 @@
+    decimals?: number
+}
+
+export function formatNumber(value: number | BigNumber | undefined, options?: FormatNumberOptions): string | undefined {
```
```suggestion
export function formatNumber(value: BigNumber.Value, options?: FormatNumberOptions): string | undefined {
    value = new BigNumber(value)
```

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,136 @@
+    return getNonHumanValue(price, 9).toNumber()
+}
+
+export function formatBigValue(
+    value?: BigNumber | number,
+    decimals = 4,
+    defaultValue = '-',
+    minDecimals: number | undefined = undefined,
+): string {
```
```suggestion
export function formatBigValue(
    value: BigNumber.Value,
    decimals = 4,
    defaultValue = '-',
    minDecimals: number | undefined = undefined,
): string {
    value = new BigNumber(value)
```

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,136 @@
+    return value?.div(getExponentValue(decimals))
+}
+
+export function getNonHumanValue(value: BigNumber | number, decimals = 0): BigNumber {
```
```suggestion
export function getNonHumanValue(value: BigNumber.Value, decimals = 0): BigNumber {
```

DCarranza committed:
feat(BarnBridge): remove a bunch of dead code

 a lot of that code is odd and unneeded => remove and only keep what I use

septs commented on packages/web3-shared/src/utils/number.ts:
```diff
@@ -22,3 +22,8 @@ export function isLessThan(a: BigNumber.Value, b: BigNumber.Value) {
 }
+
+/** x ** n */
+export function getExponentValue(decimals = 0): BigNumber {
```
plz remove the function, use `pow10` replace.

septs commented on packages/maskbook/src/plugins/BarnBridge/constants.ts:
```diff
@@ -0,0 +1,28 @@
+export const API_URL = 'https://api.barnbridge.com/'
+export const SY_URL = 'smart-yield/'
+
+export const URL_PATTERN = new RegExp(`(https)\:\/\/.*\.?${escapeRegExp(BASE_URL)}`)
```
```suggestion
export const URL_PATTERN = new RegExp(`https?\:\/\/.*\.?${escapeRegExp(BASE_URL)}`)
```

septs commented on packages/maskbook/src/plugins/BarnBridge/UI/YieldFarmingView.tsx:
```diff
@@ -0,0 +1,235 @@
+                                {t('plugin_barnbridge_yield_farming_time_left')}
+                            </Typography>
+                            <Typography className={classes.gridCellValue} variant="h5">
+                                2d 18h 42m 12s
```
why hard-coded?

DCarranza replied:
This tab is disabled. I wrote the UI for it but the data source is not hooked up. That is another feature which will happen later.

DCarranza replied:
So it's all hardcoded until then? If you prefer I parameterize it before landing i can do that.

septs commented on packages/maskbook/src/plugins/BarnBridge/UI/SmartYieldPoolsView.tsx:
```diff
@@ -0,0 +1,165 @@
+        />
+    )
+
+    const tabs = [<SYPoolsTab />, <SYWalletTab />].filter(Boolean)
```
why need `.filter`?

DCarranza replied:
not needed. removed

septs commented on packages/maskbook/src/plugins/BarnBridge/SNSAdaptor/index.tsx:
```diff
@@ -0,0 +1,44 @@
+const sns: Plugin.SNSAdaptor.Definition = {
+    ...base,
+    init(signal) {
+        console.debug('Example plugin has been loaded.')
```
plz remove `console.*`.

septs commented on packages/maskbook/src/plugins/BarnBridge/Model/SYPortfolioModel.tsx:
```diff
@@ -0,0 +1,34 @@
+
+        const realData = await response.json()
+
+        if (realData.data.length > 0) {
+            const data = realData.data[realData.data.length - 1]
+            return {
+                seniorValue: data.seniorValue,
+                juniorValue: data.juniorValue,
+            }
+        } else {
+            return {
+                seniorValue: 0,
+                juniorValue: 0,
+            }
+        }
```
```suggestion
        import { last } from 'lodash-es'

        const data = last(realData.data)
        return {
            seniorValue: data?.seniorValue ?? 0,
            juniorValue: data?.juniorValue ?? 0,
        }
```

DCarranza replied:
ah much cleaner! thanks for the suggestion

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,33 @@
+
+export function formatUSDValue(value: BigNumber.Value, decimals = 2, minDecimals = decimals): string {
+    value = new BigNumber(value)
+    if (value === undefined) {
+        return '-'
+    }
```
???

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,33 @@
+    minDecimals: number | undefined = undefined,
+): string {
+    value = new BigNumber(value)
+    if (value === undefined) {
+        return defaultValue
+    }
```
???

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,33 @@
+        return defaultValue
+    }
+
+    const bnValue = new BigNumber(value)
+
+    if (bnValue.isNaN()) {
+        return defaultValue
+    }
+
+    return new BigNumber(bnValue.toFixed(decimals)).toFormat(minDecimals)
```
```suggestion
    if (value.isNaN()) {
        return defaultValue
    }

    return new BigNumber(value.toFixed(decimals)).toFormat(minDecimals)
```

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,33 @@
+        return '-'
+    }
+
+    const val = BigNumber.isBigNumber(value) ? value : new BigNumber(value)
+    const formattedValue = formatBigValue(val.abs(), decimals, '-', minDecimals)
+
+    return val.isPositive() ? `$${formattedValue}` : `-$${formattedValue}`
```
```suggestion
    const symbol = value.isPositive() ? '$' : '-$'

    return symbol + formatBigValue(value.abs(), decimals, '-', minDecimals)
```

septs commented on packages/maskbook/src/plugins/BarnBridge/constants.ts:
```diff
@@ -0,0 +1,28 @@
+export const COLOR_SY_JUNIOR_TEXT = '#a26ee3'
+
+// Strings
+export const BB_SY_CREAM: string = 'C.R.E.A.M FINANCE'
+export const BB_SY_AAVE: string = 'AAVE'
+export const BB_SY_COMPOUND: string = 'Compound'
```
```suggestion
export const BB_SY_CREAM = 'C.R.E.A.M FINANCE'
export const BB_SY_AAVE = 'AAVE'
export const BB_SY_COMPOUND = 'Compound'
```

DCarranza committed:
feat(BarnBridge): comments v3

More comments, mostly around utils.tsx

DCarranza committed:
feat(BarnBridge): remove console.logs

remove console.logs that got left behind

septs commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/web3/utils.ts:
```diff
@@ -0,0 +1,22 @@
+    value: BigNumber.Value,
+    decimals = 4,
+    defaultValue = '-',
+    minDecimals: number | undefined = undefined,
```
```suggestion
    minDecimals?: number,
```

septs commented on packages/maskbook/src/plugins/BarnBridge/Model/SYPoolModel.tsx:
```diff
@@ -0,0 +1,64 @@
+            credentials: 'omit',
+        })
+
+        const realData = await response.json()
```
plz provide response payload typing information

```suggestion
        interface Payload<T> {
            data: T
        }
        interface Entry {
            protocolId: string
            underlyingSymbol: string
            state: State
        }
        interface State {
            seniorApy: number
            juniorApy: number
            seniorLiquidity: string
            juniorLiquidity: string
            /** etc ... **/
        }
        const payload: Payload<Entry[]> = await response.json()
```

septs reviewed (changes requested):
plz do not use `any` type.

DCarranza committed:
feat(BarnBridge): remove all usage of any

Removed all anys except for one with passing around a useStyles. Not sure how to handle that one

Jack-Works commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/PortfolioBalance.tsx:
```diff
@@ -0,0 +1,98 @@
+    progress: {
+        display: 'flex',
+        marginTop: 24,
+        '& .MuiLinearProgress-colorPrimary': {
```
Please use constants exported from `@material-ui`:

`linearProgressClasses.barColorPrimary`

All other cases are the same

Jack-Works commented on packages/maskbook/src/plugins/BarnBridge/Dashboard/index.tsx:
```diff
@@ -0,0 +1,13 @@
+import type { Plugin } from '@masknet/plugin-infra'
+import { base } from '../base'
+
+const dashboard: Plugin.Dashboard.Definition = {
```
Remove dashboard entry if you're not using it.

Jack-Works commented on packages/maskbook/src/plugins/BarnBridge/Worker/index.ts:
```diff
@@ -0,0 +1,21 @@
+// import { Ok, Some } from '@masknet/plugin-infra'
+import { base } from '../base'
+
+const worker: Plugin.Worker.Definition = {
```
Remove worker entry if you're not using it.

DCarranza committed:
feat(BarnBridge): remove any from useStyles param

Found out it's just a Record type

Jack-Works commented on packages/maskbook/src/plugins/BarnBridge/base.ts:
```diff
@@ -0,0 +1,19 @@
+        fallback:
+            'BarnBridge is a protocol that enables users to hedge against DeFi yield sensitivity and price volatility',
+    },
+    publisher: { name: { fallback: 'Mask Network' }, link: 'https://mask.io/' },
```
You should set the publisher to yourself.

DCarranza commented:
> plz do not use `any` type.

Remove all any types!

septs commented on packages/maskbook/src/plugins/BarnBridge/Model/SYPoolModel.tsx:
```diff
@@ -1,26 +1,43 @@
-        const realData = await response.json()
-        realData.data.map(function (entry: any) {
+        const payload: Payload<Entry[]> = await response.json()
+        payload.data.map(function (entry: Entry) {
```
```suggestion
        payload.data.map((entry) => {
```

septs commented on packages/maskbook/src/plugins/BarnBridge/UI/SmartYieldPoolView.tsx:
```diff
@@ -80,7 +80,7 @@ const useStyles = makeStyles()((theme) => ({
 }))
 
-const protocolNameToIcon: any = {
+const protocolNameToIcon: { [id: string]: JSX.Element } = {
```
```suggestion
const protocolNameToIcon: Record<string, JSX.Element> = {
```

DCarranza committed:
feat(BarnBridge): jack's comments

Remove worker, dash, and non-exported MaterialUI styles

DCarranza committed:
feat(BarnBridge): sept's comments

const protocolNameToIcon: Record<string, JSX.Element> = {

Jack-Works commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/PortfolioBalance.tsx:
```diff
@@ -0,0 +1,97 @@
+            <Divider />
+            <LinearProgress
+                className={classes.progress}
+                classes={{ barColorPrimary: COLOR_SY_SENIOR_TEXT }}
```
Hmm does this really works? It means add class name `#00d395` to `bar` with color primary.

DCarranza replied:
Yeah I'm not sure -- the barnbridge API is broken so i can't test. :\

DCarranza replied:
Reverted back for now

Jack-Works commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/PortfolioBalance.tsx:
```diff
@@ -0,0 +1,97 @@
+    return (
+        <div className="card">
+            <div className="card-header">
+                <Typography variant="body1" color="#fff">
```
Does those hard-coded styles work in both light and dark mode?

DCarranza replied:
I'm not supporting both light and dark mode: looking at other plugins most don't either. Here's pool together
<img width="544" alt="Screen Shot 2021-09-16 at 8 29 24 PM" src="https://user-images.githubusercontent.com/3681070/133719795-d3062737-0b98-4882-b8b8-5c685c10ef28.png">
<img width="540" alt="Screen Shot 2021-09-16 at 8 29 34 PM" src="https://user-images.githubusercontent.com/3681070/133719796-5168c393-c2ce-45f3-b5ce-b7144c1cc5b2.png">

Jack-Works reviewed (changes requested):
Request review from @guanbinrui, request test from @DimensionDev/qa

guanbinrui commented on packages/maskbook/src/plugins/BarnBridge/BarnBridgeIcon.tsx:
```diff
@@ -0,0 +1,184 @@
+export const BarnBridgeYieldFarmingIcon: FC<SvgIconProps> = (props) => (
+    <SvgIcon {...props}>
+        <svg width="50" height="50" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <circle cx="5.369" cy="16.631" r="4.369" stroke="currentColor" stroke-width="1.5" />
```
```suggestion
            <circle cx="5.369" cy="16.631" r="4.369" stroke="currentColor" strokeWidth="1.5" />
```

DCarranza replied:
Copied this from OMGSVG as stated in an earlier comment

DCarranza replied:
but i'll change it!

guanbinrui replied:
Yes, it's needed, also other properties. Otherwise, the linter will come to you. 😂

DCarranza replied:
replaced all iterations in this file

guanbinrui commented on packages/maskbook/src/plugins/BarnBridge/BarnBridgeIcon.tsx:
```diff
@@ -0,0 +1,184 @@
+            <path
+                d="M18.8 13.2V3.8a3 3 0 0 0-3-3H4.2a3 3 0 0 0-3 3v16.4a3 3 0 0 0 3 3h7M6.8 8.8h6.4M6.8 12h2.4"
+                stroke="currentColor"
+                stroke-linecap="round"
```
```suggestion
                strokeLinecap="round"
```

DCarranza replied:
replaced all instances

guanbinrui commented on packages/maskbook/src/plugins/BarnBridge/SNSAdaptor/index.tsx:
```diff
@@ -0,0 +1,41 @@
+        <MaskbookPluginWrapper pluginName="BarnBridge">
+            <Suspense fallback={<SnackbarContent message="Mask is loading this plugin..." />}>
+                <EthereumChainBoundary
+                    chainId={ChainId.Mainnet}
```
It should read the current chain id instead of dead-code Mainnet. If given an invalid chain id, the boundary will reveal the switch chain panel for you.

DCarranza replied:
Ah will do good callout. This plugin doesn't really use any contract info -- although i guess it does use wallet address. I'll do it the right way

DCarranza replied:
So taking a look at it -- `PoolTogether` also hardcodes the chain id. Since I use no web3 calls in the entire plugin i don't see a reason why it makes sense to do this. For a future update I will need to (for the YieldFarming tab) when that features come i can integrate web3 and the token contracts for the correct chain support.

guanbinrui commented on packages/maskbook/src/plugins/BarnBridge/UI/SmartYieldPoolView.tsx:
```diff
@@ -0,0 +1,259 @@
+    const { classes } = useStyles()
+    const { t } = useI18N()
+
+    const poolURL = ''
```
Is it a dangling variable?

DCarranza replied:
Ah yeah it is good catch

guanbinrui commented on packages/maskbook/src/plugins/BarnBridge/UI/SmartYieldPoolView.tsx:
```diff
@@ -0,0 +1,259 @@
+            <GenerateProtocolName protocolName={props.protocolName} />
+            <GenerateHeader />
+            {props.coins.map((coin) => (
+                <GeneratePoolCell
```
The `key` was lost here.

DCarranza replied:
I'm not sure what you mean here.... `key` in this case in `coin` and it's used to populate the GeneratePoolCell function

DCarranza replied:
You answered below -- thanks :)

guanbinrui commented on packages/maskbook/src/plugins/BarnBridge/constants.ts:
```diff
@@ -0,0 +1,28 @@
+export const BASE_URL = 'barnbridge.com'
+export const APP_URL = 'https://app.barnbridge.com/'
+export const API_URL = 'https://api.barnbridge.com/'
+export const SY_URL = 'smart-yield/'
```
Not a valid URL?

DCarranza replied:
I can rename it to `SY_URL_FRAGMENT` since it's not technically a valid url

guanbinrui commented on packages/maskbook/src/plugins/BarnBridge/UI/SmartYieldPoolsView.tsx:
```diff
@@ -0,0 +1,170 @@
+    return (
+        <div className={classes.root}>
+            {protocols.map((name: string) => (
+                <SmartYieldPoolView protocolName={name} coins={syData[name]} />
```
The `key` was lost here.

DCarranza replied:
Again not too sure what this means. Is there some scope issue where the key is invalid?

guanbinrui replied:
I mean this [Lists and Keys](https://reactjs.org/docs/lists-and-keys.html#keys).

DCarranza replied:
Ah cool i never knew this exist -- will add the keys!

guanbinrui commented on packages/maskbook/src/plugins/BarnBridge/Model/SYPoolModel.tsx:
```diff
@@ -0,0 +1,81 @@
+
+export function SmartYieldPoolModelGetData() {
+    return useAsyncRetry(async () => {
+        const response = await fetch('https://api.barnbridge.com/api/smartyield/pools?originator=all', {
```
Move this external API request to the worker module. The SNS provider may enable the CSP, which means external requests are not allowed.

DCarranza replied:
Interesting -- any examples of what the worker module looks like; sorry i'm not too familiar with it

guanbinrui replied:
Please ref to the Gitcoin plugin as an example. You can find it here in `packages/maskbook/src/plugins/Gitcoin/index.ts`.

DCarranza replied:
I've spent a few hours on this and I can't figure it out. My objects never are populated; I need some help understanding what I'm doing wrong. Sending out a push

guanbinrui commented on packages/maskbook/src/plugins/BarnBridge/Model/SYPoolModel.tsx:
```diff
@@ -0,0 +1,81 @@
+import { BB_SY_CREAM, BB_SY_AAVE, BB_SY_COMPOUND, APP_URL, SY_URL } from '../constants'
```
It should be a `.ts` file.

nuanyang233 commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/PortfolioBalance.tsx:
```diff
@@ -0,0 +1,97 @@
+import { useI18N } from '../../../../utils/i18n-next-ui'
+import { COLOR_SY_SENIOR_TEXT, COLOR_SY_JUNIOR_TEXT } from '../../constants'
+
+const s = makeStyles()((theme) => ({
```
I don't think `s` is a good name

nuanyang233 commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/PortfolioBalance.tsx:
```diff
@@ -0,0 +1,97 @@
+    dataColumn: {
+        paddingLeft: 16,
+        position: 'relative',
+        backgroundColor: 'var(--color)',
```
`var(-color)` is mysterious to me 👀 . Where is `--color` from.

nuanyang233 commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/PortfolioBalance.tsx:
```diff
@@ -0,0 +1,97 @@
+    },
+}))
+
+type PortfolioPropsLabel = [label: string, value: number]
```
You can just define it as [Tuple](https://www.typescriptlang.org/docs/handbook/2/objects.html#tuple-types).

nuanyang233 commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/PortfolioBalance.tsx:
```diff
@@ -0,0 +1,97 @@
+        data: [[label1, value1], [label2, value2]],
+    } = props
+
+    const progress = ((value1 ?? 0) * 100) / ((value1 ?? 0) + (value2 ?? 0))
```
When the denominator is 0 you get `Infinity`

DCarranza replied:
Grabbed off the bb frontend. Just tested it out Infinity passed into the progress bar doesn't break it. It just defaults to 100

nuanyang233 commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/PortfolioBalance.tsx:
```diff
@@ -0,0 +1,97 @@
+                    {t('plugin_barnbridge_sy_portfolio_balance')}
+                </Typography>
+            </div>
+            <div className="p-24 flexbox-grid flow-col gap-16">
```
Where are these classes defined?

DCarranza replied:
I grabbed this wholesale from barnbridge code -- I can remove it/clean it up

nuanyang233 commented on packages/maskbook/src/plugins/BarnBridge/Model/SYPortfolioModel.tsx:
```diff
@@ -0,0 +1,31 @@
+
+export function SmartYieldPortfolioModelGetData() {
+    return useAsyncRetry(async () => {
+        const walletAddress = currentAccountSettings.value
```
Please subscribe to this setting with `useValueRef`.

DCarranza committed:
feat(BarnBridge): switch to Worker

Use worker module for network requests

DCarranza commented on packages/maskbook/src/plugins/BarnBridge/UI/SmartYieldPoolsView.tsx:
```diff
@@ -0,0 +1,176 @@
+    const walletAddress = useValueRef(currentAccountSettings)
+
+    //#region data
+    const { value: syData, error: error, loading: loadingPools, retry: retry } = useSYPoolData()
```
after switching over to worker these syData is never populated ->  I have no idea why

DCarranza commented on packages/maskbook/src/plugins/BarnBridge/UI/SmartYieldPoolsView.tsx:
```diff
@@ -0,0 +1,176 @@
+    //#region data
+    const { value: syData, error: error, loading: loadingPools, retry: retry } = useSYPoolData()
+    const {
+        value: syPortfolioData,
```
Same here -- this is never populated. Take a look at useSYPortfolioData, it's pretty straight forward and should fetch the requested data. Please help!

septs commented on packages/maskbook/src/plugins/BarnBridge/Model/SYPoolModel.ts:
```diff
@@ -0,0 +1,81 @@
+
+export function SmartYieldPoolModelGetData() {
+    return useAsyncRetry(async () => {
+        const response = await fetch('https://api-v2.barnbridge.com/api/smartyield/pools?originator=all', {
```
```suggestion
        const response = await fetch(urlcat(API_URL, '/api/smartyield/pools', { originator: 'all' }), {
```

DCarranza replied:
Done! @septs  if you can figure out why in the world my transition to Worker module isn't working I'll love you forever

septs commented on packages/maskbook/src/plugins/BarnBridge/Model/SYPortfolioModel.tsx:
```diff
@@ -0,0 +1,29 @@
+export function SmartYieldPortfolioModelGetData(walletAddress: string) {
+    return useAsyncRetry(async () => {
+        const response = await fetch(
+            `https://api-v2.barnbridge.com/api/smartyield/users/${walletAddress}/portfolio-value`,
```
```suggestion
            urlcat(API_URL, '/smartyield/users/:walletAddress/portfolio-value', { walletAddress }),
```

septs commented on packages/maskbook/src/plugins/BarnBridge/constants.ts:
```diff
@@ -0,0 +1,28 @@
+
+export const BASE_URL = 'barnbridge.com'
+export const APP_URL = 'https://app.barnbridge.com/'
+export const API_URL = 'https://api-v2.barnbridge.com/'
```
```suggestion
export const API_URL = 'https://api-v2.barnbridge.com/api/'
```

septs commented on packages/maskbook/src/plugins/BarnBridge/apis/index.ts:
```diff
@@ -0,0 +1,104 @@
+export type SYPoolModelData = { [id: string]: SYCoinProps[] }
+
+export async function SYGetPools() {
+    const response = await fetch('https://api-v2.barnbridge.com/api/smartyield/pools?originator=all', {
```
```suggestion
    const response = await fetch(urlcat(API_URL, '/smartyield/pools', { originator: 'all' }), {
```

septs commented on packages/maskbook/src/plugins/BarnBridge/apis/index.ts:
```diff
@@ -0,0 +1,104 @@
+
+export async function SYGetPortfolio(walletAddress: string) {
+    const response = await fetch(
+        `https://api-v2.barnbridge.com/api/smartyield/users/${walletAddress}/portfolio-value`,
```
```suggestion
        urlcat(API_URL, '/smartyield/users/:walletAddress/portfolio-value', { walletAddress }),
```

DCarranza committed:
feat(BarnBridge): remove naughty words

remove naughty words + switch to urlcat for portfolio

septs commented on packages/maskbook/src/plugins/BarnBridge/apis/index.ts:
```diff
@@ -84,15 +84,12 @@ function PrettifyLiquidity(liquidity: string) {
-            credentials: 'omit',
-        },
-    )
+    const response = await fetch(urlcat(API_URL, '/api/smartyield/pools', { originator: 'all' }), {
```
move `/api/` prefix string to `API_URL` constant, thanks

DCarranza replied:
moved

DCarranza committed:
feat(barnbridge): remove all fetch w/o urlcat

missing a few and migrate /api/ to the constant

DCarranza commented:
Soft ping :)

DCarranza commented:
@guanbinrui

septs commented:
> Soft ping :)

plz wait to 2021/9/23 10:00 AM after (UTC+8).

DCarranza commented:
Ah my bad! Didn't know what time zone you were all in :)

guanbinrui commented on packages/maskbook/src/plugins/BarnBridge/BarnBridgeIcon.tsx:
```diff
@@ -0,0 +1,184 @@
+    <SvgIcon {...props}>
+        <svg width="30" height="30" viewBox="0 0 50 50" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <linearGradient id="a" x1="34.912" x2="5.152" y1="7.453" y2="32.493" gradientUnits="userSpaceOnUse">
+                <stop stop-color="#B6509E" />
```
```suggestion
                <stop stopColor="#B6509E" />
```

guanbinrui commented on packages/maskbook/src/plugins/BarnBridge/BarnBridgeIcon.tsx:
```diff
@@ -0,0 +1,184 @@
+        <svg width="30" height="30" viewBox="0 0 50 50" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <linearGradient id="a" x1="34.912" x2="5.152" y1="7.453" y2="32.493" gradientUnits="userSpaceOnUse">
+                <stop stop-color="#B6509E" />
+                <stop offset="1" stop-color="#2EBAC6" />
```
```suggestion
                <stop offset="1" stopColor="#2EBAC6" />
```

guanbinrui commented on packages/maskbook/src/plugins/BarnBridge/Import/BarnBridgeFrontEnd/PortfolioBalance.tsx:
```diff
@@ -0,0 +1,102 @@
+                </Typography>
+            </div>
+            <div>
+                <div>
```
Remove this?

guanbinrui commented on packages/maskbook/src/plugins/BarnBridge/UI/SmartYieldPoolView.tsx:
```diff
@@ -0,0 +1,252 @@
+}
+
+function onClickPoolCell(clickUrl: string) {
+    window.open(clickUrl)
```
```suggestion
    window.open(clickUrl, '_blank', 'noopener noreferrer')
```

guanbinrui commented on packages/maskbook/src/plugins/BarnBridge/index.ts:
```diff
@@ -0,0 +1,17 @@
+        hotModuleReload: (hot) =>
+            import.meta.webpackHot && import.meta.webpackHot.accept('./SNSAdaptor', () => hot(import('./SNSAdaptor'))),
+    },
+    Dashboard: undefined,
```
```suggestion
```

guanbinrui commented on packages/maskbook/src/plugins/BarnBridge/apis/index.ts:
```diff
@@ -0,0 +1,101 @@
+    return Math.round(intLiquidity).toString()
+}
+
+export async function SYGetPortfolio(walletAddress: string) {
```
The `walletAddress` is not used.

guanbinrui commented on packages/maskbook/src/plugins/BarnBridge/apis/index.ts:
```diff
@@ -0,0 +1,101 @@
+function PrettifyLiquidity(liquidity: string) {
+    const intLiquidity = Number.parseInt(liquidity, 10)
+    if (intLiquidity >= 1000000) {
+        return Math.round(intLiquidity / 1000000).toString() + 'M'
```
```
const MILLION = 1000000
const THOUSAND = 1000
```

septs commented on packages/maskbook/src/plugins/BarnBridge/Model/SYPoolModel.ts:
```diff
@@ -0,0 +1,81 @@
+}
+
+function PrettifyLiquidity(liquidity: string) {
+    const intLiquidity = Number.parseInt(liquidity, 10)
+    if (intLiquidity >= 1000000) {
+        return Math.round(intLiquidity / 1000000).toString() + 'M'
+    }
+    if (intLiquidity >= 1000) {
+        return Math.round(intLiquidity / 1000).toString() + 'K'
+    }
+    return Math.round(intLiquidity).toString()
```
```suggestion
    const value = Number.parseInt(liquidity, 10)
    const unit = Math.log(value) / Math.log(1000)
    const symbols = ['', 'k', 'M', 'G', 'T'] // see https://en.wikipedia.org/wiki/Metric_prefix
    return `${Math.round(value / (1000 ** unit))}${symbols[unit]}`
```

(The code need verify)

septs replied:
plz use this algorithm simplify the logic, thanks

DCarranza commented:
Btw using the Worker module broke network connections for me -> await PluginBarnBridgeRPC.SYGetPools() and await PluginBarnBridgeRPC.SYGetPortfolio(walletAddress) do not work => they never return data. Something is super up with using the Worker module y'all suggested up above. I tried debugging but have no idea what's up

DCarranza committed:
feat(BarnBridge): comments

comments

DCarranza committed:
feat(BarnBridge): urlcat

urcat swap

DCarranza committed:
feat(BarnBridge): sypoolmodel

sy pool model changes

DCarranza committed:
feat(BarnBridge): more comments

more comments

guanbinrui commented:
Hi @DCarranza, would you please resolve those conflicts? Both ping @DimensionDev/qa.

DCarranza commented:
Rebase + conflict fixes

AinuX commented:
1. Need to adapt twitter light mode
![image](https://user-images.githubusercontent.com/2582974/134653490-1e513b86-2ade-4e1c-bd7e-bb056b79deb3.png)
2. It's better to freeze table header

AinuX commented:
3. Different chain should show different mark info. 
Currently, we always show market info under ETH chain even switch to Polygon
![image](https://user-images.githubusercontent.com/2582974/134658265-5cef9b5e-9342-4fda-9b95-8d08949ea0df.png)
![image](https://user-images.githubusercontent.com/2582974/134658302-8544925a-fe11-4dbe-b859-0eb5b8f41f6e.png)

AinuX commented:
4. Accuracy
![image](https://user-images.githubusercontent.com/2582974/134659420-89cf60b7-edd1-40fe-b222-e4e709600da7.png)
![image](https://user-images.githubusercontent.com/2582974/134659472-125c7e29-3cbf-4770-9c75-d849a16151be.png)

DCarranza committed:
feat(BarnBridge): add polygon support

Add polygon support for Smart Yield

DCarranza committed:
feat(BarnBridge): light/dark support

add twitter dark mode and light mode support.

septs commented on packages/maskbook/src/plugins/BarnBridge/apis/index.ts:
```diff
@@ -86,15 +105,19 @@ function ConstructRedirectUrl(protocolName: string, coin: string) {
 }
 
 function PrettifyLiquidity(liquidity: string) {
```
why revert the function?

DCarranza replied:
Ah i originally thought it was causing truncation issues but it wasn't the root cause of the accuracy problems - i meant to restore this back to the log algo. Let me push real quick

DCarranza committed:
feat(BarnBridge): restore Prettify

restore prettifyLiquidity to log algo

DCarranza commented:
Alrighty next step is the deposit dialog. That'll take me a few days

DCarranza commented:
So these last pushes added in:
1) Polygon support
2) Light/Dark mode support
3) Rounding/truncation issue fix ('accuracy' problem mentioned earlier)

guanbinrui commented on packages/maskbook/src/plugins/BarnBridge/hooks/useModels.ts:
```diff
@@ -0,0 +1,17 @@
+
+export function useSYPortfolioData(chainId: ChainId, walletAddress: string) {
+    return useAsyncRetry(async () => {
+        const data = await SYGetPortfolio(chainId, walletAddress)
```
It should fetch data from the background with the internal RPC bridge.

`PluginBarnBridgeRPC.SYGetPortfolio`

guanbinrui commented:
Hi @DCarranza, any updates? It's iime to confirm those issues?

Tedko commented:
Hi @DCarranza keep us updated thx
